src/Randomizer.SMZ3.Tracking/AutoTracking/GameService.cs

```diff
@@ -112,14 +127,14 @@ public bool TryGiveItems(List<Item> items, int fromPlayerId)
         /// <returns>False if it is currently unable to give an item to the player</returns>
         public bool TryHealPlayer()
         {
-            if (_autoTracker == null || !_autoTracker.IsConnected || !_autoTracker.IsInSMZ3)
+            if (AutoTracker == null || !AutoTracker.IsConnected || !AutoTracker.IsInSMZ3)
```

Comment:
Maybe change this to `AutoTracker == null || AutoTracker is not { IsConnected: true, IsInSMZ3: true })` for consistency with the above? I think the pattern matching you did above is cleaner.